base/applications/network/nslookup/nslookup.c

```diff
@@ -812,15 +812,16 @@ int main( int argc, char* argv[] )
 
             return -2;
         }
-    }
 
-    strncpy( State.domain, pNetInfo->DomainName, 255 );
-    strncpy( State.srchlist[0], pNetInfo->DomainName, 255 );
-    strncpy( State.DefaultServerAddress,
-             pNetInfo->DnsServerList.IpAddress.String,
-             15 );
+        strncpy( State.domain, pNetInfo->DomainName, 255 );
+        strncpy( State.srchlist[0], pNetInfo->DomainName, 255 );
+        strncpy( State.DefaultServerAddress,
+                 pNetInfo->DnsServerList.IpAddress.String,
+                 15 );
+
+        HeapFree( ProcessHeap, 0, pNetInfo );
+    }
 
```

Comment:
Shouldn't we just exit the program if GetNetworkParams returned something else than ERROR_BUFFER_OVERFLOW?

Reply:
Probably a sane decision. I have a hard time imagining a scenario in which it is possible to send DNS requests, but retrieving the network parameters of the *local* machine fails. Please see the new version of the fix.